app/src/main/java/org/fossasia/openevent/app/core/event/create/CreateEventViewModel.java

```diff
@@ -56,39 +67,51 @@ public CreateEventPresenter(EventRepository eventRepository, CurrencyUtils curre
         countryCurrencyMap = currencyUtils.getCountryCurrencyMap();
         countryList = new ArrayList<>(countryCurrencyMap.keySet());
         currencyCodesList = currencyUtils.getCurrencyCodesList();
+
+        setPaymentPreferences(preferences);
+
     }
 
-    @Override
-    public void start() {
-        getView().attachCountryList(countryList, getCountryIndex());
-        getView().attachCurrencyCodesList(currencyCodesList);
+    public int setTimeZoneList(List<String> timeZoneList) {
+        return timeZoneList.indexOf(TimeZone.getDefault().getID());
+    }
 
-        List<String> timeZoneList = getView().getTimeZoneList();
-        getView().setDefaultTimeZone(
-            timeZoneList.indexOf(TimeZone.getDefault().getID())
-        );
+    public List<String> getCountryList() {
+        return countryList;
+    }
 
-        setPaymentPreferences(preferences);
+    public List<String> getCurrencyCodesList() {
+        return currencyCodesList;
     }
 
     public Event getEvent() {
         return event;
     }
 
+    public LiveData<Event> getEventLiveData() {
+        eventMutableLiveData.setValue(event);
+        return eventMutableLiveData;
+    }
+
     private boolean verify() {
+        if (event.getName() == null) {
+            onError.setValue("Event Name cannot be empty");
```

Comment:
This check is already done in layout file by `validateEmpty=true`. So, this shouldn't be needed.